src/hazmat/mod.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! ⚠️ Low-level "hazmat" functions.
+//!
+//! This module contains low level APIs that should *not* be used or needed by
+//! most users.
+//!
+//! These functions are exported to aid very advanced use cases.
+
+#![cfg(feature = "low-level-api")]
+
+pub mod olm;
```

Comment:
I wonder if this is a bit much considering that there's a single type in the olm module. You plan on having a megolm variant of the message key as well? Not really important to change this if you don't feel like it.

Reply:
I'm persuadable, but my main thought was that a lot of the types in `olm` aren't obviously named in an olm specific way, e.g. `MessageKey` could equally by related to megolm or whatever. :shrug:

Reply:
Yeah, let's leave it.